docs/format_description.rst

```diff
@@ -291,6 +292,28 @@ regardless of format.
       "source" : "[some nbformat output text]"
     }
 
+
+Cell attachments
+----------------
+Markdown and raw cells can have a number of attachments, typically inline
```

Comment:
Let's add a clear, explicit statement that `attachments` may be undefined, and add a

```restructuredtext
.. versionadded:: 4.1
```